apps/geo/serializers.py

```diff
@@ -148,3 +153,38 @@ class Meta:
             'region_title', 'admin_level_level', 'admin_level_title',
             'parent'
         )
+
+
+class RegionGqSerializer(ProjectPropertySerializerMixin, UserResourceSerializer):
+    project = IntegerIDField()
+
+    class Meta:
+        model = Region
+        exclude = ('geo_options', 'is_published')
+
+    def validate_project(self, project):
+        try:
+            project = Project.objects.get(id=project)
+        except Project.DoesNotExist:
+            raise serializers.ValidationError(
+                'Project matching query does not exist'
+            )
+
+        if not project.can_modify(self.context['request'].user):
+            raise serializers.ValidationError('Permission Denied')
+        return project.id
```

Comment:
Better Check the Project and return Object if this field `project` is Mandatory in client.
If this field is not mandatory, You need to actually check the `validated_data` has project or not.